src/client/directory.ts

```diff
@@ -10,60 +10,150 @@
  */
 
 /**
- * Type definitions that is are wherever the API expects the ID or of an application to be passed. We have both, since
- * internally appId is used to identify app, but externally we use name to repsect specification
+ * Type alias to indicate when an Application Identifier should be passed. Application Identifiers
+ * are described [here](https://fdc3.finos.org/docs/1.0/appd-discovery#application-identifier).
  *
- * It is not possible for TypeScript to verify that only valid application names/appIds are passed
- * to the FDC3 API. This type definition is more a hint to the callee, over a plain "string" argument.
+ * This type alias exists to disambiguate raw string app identity from the [[AppName]].
  */
 export type AppId = string;
+/**
+ * Type alias to indicate when an Application Name should be passed. This will be a human-readable
+ * name that will be displayed on screen.
+ *
+ * This type alias exists to disambiguate raw string app identity from the [[AppId]].
+ */
 export type AppName = string;
 
 /**
- * An application in the app directory
+ * An application in the app directory.
  */
 export interface Application {
+    /**
+     * The Application Identifier. Please see https://fdc3.finos.org/docs/1.0/appd-discovery#application-identifier.
+     */
     appId: AppId;
+    /**
+     * The human readable app name.
+     */
     name: AppName;
+    /**
+     * An application manifest, used to launch the app. This should be a URL that points to an OpenFin json manifest.
+     */
     manifest: string;
+    /**
+     * The manifest type. Always 'openfin'.
+     */
     manifestType: string;
-
+    /**
+     * The version of the app. Please use [semantic versioning](https://en.wikipedia.org/wiki/Software_versioning).
+     */
     version?: string;
+    /**
+     * The human-readable title of the app, typically used by the launcher UI. If not provided, the name is used.
+     */
     title?: string;
+    /**
+     * Tooltip used by any UIs that display app information. A short explanatory text string.
+     */
     tooltip?: string;
+    /**
+     * Longer description of the app.
+     */
     description?: string;
+    /**
+     * Images that can be displayed as part of the app directory entry. Use these for screenshots, previews or similar. These are not the
+     * application icons: use 'icons' for that.
+     */
     images?: AppImage[];
+    /**
+     * Contact email address.
+     */
     contactEmail?: string;
+    /**
+     * Support email address.
```

Comment:
This comment seems a bit unnecessary as it doesn't really give any more info than the property name. If we're going to comment this kind of thing can we add a bit more depth.

Reply:
Disagree that it's lose or expand. Sometimes there isn't that much to say, but for the docs we want *something*

Reply:
I reckon I can be a bit less lazy though

Reply:
@bryangaleOF, I would say a property called "supportEmail" is pretty self-descriptive and adding a comment saying "Support Email" doesn't really add anything of any value. If we're going to have doc comments on every property they should at least form a full sentence, even it if doesn't say much more in actuality.